### PR TITLE
Issue #685 - refactor common CDI code

### LINKED ISSUE #685: CDI 2.0: refactor web, ejb and jsf auto-feature bundles to make code common
A handful of weld classes have moved and changed meaning that the code can not be 100% shared. However, a good portion is probably still the same so it should be refactored to avoid duplication.

### PR DESCRIPTION
**Pull request by tevans78**
fixes #685 
#build

### CODE REVIEW

File: dev/com.ibm.ws.cdi.ejb/resources/com/ibm/ws/cdi/ejb/impl/resources/CDI.nlsprops [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Messages used by JCDI
+#-----------------------------------------------------------------------------------------------------------------------------
+
+transactional.annotation.on.ejb.CWOWB2000E=CWOWB2000E: The annotation {0} is not allowed on EJB: {1}
+transactional.annotation.on.ejb.CWOWB2000E.explanation=Annotation transactional cannot be used with EJB
+transactional.annotation.on.ejb.CWOWB2000E.useraction=The TransactionAttribute should be used instead of Transactional with EJB's
+
+lifecycle.interceptor.exception.CWOWB2001E=CWOWB2001E: A {0} lifecycle inteceptor threw an exception: {1}
+lifecycle.interceptor.exception.CWOWB2001E.explanation=Lifecycle interceptors should avoid throwing checked exceptions 
+lifecycle.interceptor.exception.CWOWB2001E.useraction=Check the logs to determine the problem
```
Comment (rhpeters) [inline dev/com.ibm.ws.cdi.ejb/resources/com/ibm/ws/cdi/ejb/impl/resources/CDI.nlsprops]: Add punctuation: Check the logs to determine the problem.

**Comment by tevans78** [comment]
@rhpeters @bcpswope I'm not making NLS changes in this PR, just moving everything around. The particular message you've picked up on is relatively old. NLS doesn't even work for CDI at the moment and I'm intending to sort that out under another issue.

**Comment by bcpswope** [comment]
@tevans78 thank you for letting us know.

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.ws.cdi.1.2.weld/bnd.bnd [hunk 2]
```diff
@@ -13,19 +13,18 @@ bVersion=1.0
...
+Bundle-Description:CDI 1.2 Weld Integration, version ${bVersion}
 
-Bundle-Description:cdi 1.2 weld impl, version ${bVersion}
+Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.7))"
 
 Export-Package: com.ibm.ws.cdi.executor, \
                 com.ibm.ws.cdi.utils
                 
-Import-Package: org.jboss.weld.*;version="[2,3)", \
+Import-Package: org.jboss.weld.*;version="[2.4.3,3.0.0)", \
```
[the remark this answers is not in the document]
Comment (tevans78) [reply dev/com.ibm.ws.cdi.1.2.weld/bnd.bnd]: fixed ... that bundle doesn't actually need an explicit Import statement any more.

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.ws.cdi.ejb/bnd.bnd [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                javax.decorator;version="[1.1,3)",\
+                javax.enterprise.context;version="[1.1,3)",\
+                javax.enterprise.context.spi;version="[1.1,3)",\
+                javax.enterprise.inject;version="[1.1,3)",\
+                javax.enterprise.inject.spi;version="[1.1,3)",\
+                com.ibm.ws.cdi.internal.interfaces, \
+                org.jboss.weld;version="[2.4,4)",\
+                org.jboss.weld.bean;version="[2.4,4)",\
+                org.jboss.weld.construction.api;version="[2.4,4)",\
+                !org.jboss.weld.ejb,\
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.ejb/bnd.bnd]: @Emily-Jiang why do I have to do this?!

Comment (Emily-Jiang) [reply dev/com.ibm.ws.cdi.ejb/bnd.bnd]: The reason is that it generates based on the bundle in the buildpath and then worked out the version range. The version range upper bound will be 3 not 4 based on the versioning scheme.

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
Failing NLS tests have now passed in this build:
https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_zrQasMYHEeeBHLpwjQKQLg

**Pull request merged**